rum/app.py

```diff
@@ -10,41 +11,40 @@ def __init__(self, username, email, access_level='read_only', send_mail=True):
         self.access_level = access_level
         self.sendmail = send_mail
         self.redshift = Redshift()
+        self.check_read_only_exists = open('sqls/check_read_only_exists.sql').read()
+        self.create_user = open('sqls/create_user.sql').read()
+        self.remove_user = open('sqls/remove_user.sql').read()
+        self.add_user_read_only_group = open('sqls/add_user_read_only_group.sql').read()
 
-        # Initial Setup
         if not self._read_only_group_exists():
             self.redshift.query('CREATE GROUP read_only;')
 
     def _read_only_group_exists(self):
-        sql = open('sqls/check_read_only_exists.sql').read()
+        sql = self.check_read_only_exists
         return self.redshift.query(sql).fetchone()[0]
 
     def add_user(self):
         """
         Add user to redshift, with random password and send email with connection data.
         :return: None
         """
-        # Create random password for user
         password = Password().str
 
-        # Create user
-        sql = open('sqls/create_user.sql').read().format(self.username, password)
+        sql = self.create_user.format(self.username, password)
         self.redshift.query(sql)
         print('User {} created with password "{}" (without quotes).'.format(self.username, password))
 
-        # Send email to user
         if self.sendmail:
             sendgrid = SendGrid(self.email)
             try:
                 response = sendgrid.send_usercreated_mail(self.username, password)
-            except Exception as error:
+            except UnauthorizedError as error:
                 raise Exception('If you do not wish to send email to users, '
-                                'then set RedshiftUserManagement(send_mail=False).'
+                                'then set RedshiftUserManagement(send_mail=False). '
                                 'Otherwise, set the correct environmental variables.', error)
             if response >= 300:
                 print('Failed to send email')
```

Comment:
You could create a method to send an email and invoke it here.

Reply:
added this exception and if clause to send_mail.py